usim/_basics/resource.py

```diff
@@ -0,0 +1,208 @@
+from typing import TypeVar, Generic, Optional, Type
+
+from .._core.loop import __LOOP_STATE__
+from ._resource_level import __specialise__, ResourceLevels
+from .tracked import Tracked
+
+
+T = TypeVar('T')
+
+
+class BaseResources(Generic[T]):
+    """
+    Internal base class for resource types
+    """
+    _levels_type = None  # type: Type[ResourceLevels[T]]
+    _available = None  # type: Tracked[ResourceLevels[T]]
+
+    @property
+    def levels(self) -> ResourceLevels[T]:
```

Comment:
Why does a resource have several levels? I would expect only one.

Reply:
Since resources consist of several distinct names->values, it technically has several levels.